mafContrib/almPlots.py

```diff
@@ -0,0 +1,223 @@
+#####################################################################################################
+# Purpose: plot skymaps/cartview plots corresponding to alms with specfied l-range (s).
+
+# Humna Awan: humna.awan@rutgers.edu
+# Last updated: 06/27/16
+ #####################################################################################################
+
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import healpy as hp
+
+__all__= ['almPlots']
+    
+def almPlots(path, outDir, bundle,
+             nside= 128, lmax= 500, filterband= 'i',
+             raRange= [-50,50], decRange= [-65,5],
+             subsetsToConsider= [[130,165], [240, 300]],
+             showPlots= True):
+    """
+
+    Plot the skymaps/cartview plots corresponding to alms with specified l-ranges. 
+    Automatically creates the output directories and saves the plots.
+
+    Required Parameters
+    -------------------
+      * path: str: path to the main directory where output directory is saved
+      * outDir: str: name of the main output directory
+      * bundle: metricBundle object.
+    
+    Optional Parameters
+    -------------------
+      * nside: int: HEALpix resolution parameter. Default: 128
+      * lmax: int: upper limit on the multipole. Default: 500
+      * filterBand: str: any one of 'u', 'g', 'r', 'i', 'z', 'y'. Default: 'i'
+      * raRange: float array: range of right ascention (in degrees) to consider in cartview plot; only useful when 
+                              cartview= True. Default: [-50,50]
+      * decRange: float array: range of declination (in degrees) to consider in cartview plot; only useful when 
+                               cartview= True. Default: [-65,5]
+      * subsetsToConsider: array of int arrays: l-ranges to consider, e.g. use [[50, 100]] to consider 50<l<100.
+                                                Currently built to handle five subsets (= number of colors built in).
+                                                Default: [[130,165], [240, 300]]
+      * showPlots: boolean: set to True if want to show figures. Default: True
+
+    """
+    # set up the output directory
+    outDir2= 'almAnalysisPlots_' + str(raRange[0]) + '<RA<' + str(raRange[1])+ '_' + str(decRange[0]) + '<Dec<' + str(decRange[1])
+    os.chdir(path + outDir)
+    if not os.path.exists(outDir2):
+        os.makedirs(outDir2)
+    
+    os.chdir(path + outDir + '/' + outDir2)  
+    outDir3= 'almSkymaps'
+    if not os.path.exists(outDir3):
+        os.makedirs(outDir3)
+        
+    os.chdir(path + outDir + '/' + outDir2)
+    outDir4= 'almCartviewMaps'
+    if not os.path.exists(outDir4):
+        os.makedirs(outDir4)
+    
+    os.chdir(path + outDir + '/' + outDir2)
+
```

Comment:
instead of "path + outDir + '/' + outDir2", suggest using 
os.path.join(path, outDir, outDir2)